### ISSUE TITLE
Improve optimization

### ISSUE THREAD

**Issue by DiMarzioBian** [FDUDSDE/CoPE#1]
Dear Author,

Hello, I am trying to test your code of CoPE. However while I am running codes on Garden dataset, the model converges fast at 2nd epoch reaching 0.078 as MRR and 0.1747 on Recall. Could you kindly give me some advice on how to further tuning the model parameters?

**Comment by yzhang1918** [comment]
Hi DiMarzioBian,

It is easy to overfit Garden. Raise weight decay to 0.01 should reproduce the results.

Thanks!

**Comment by DiMarzioBian** [comment]
Thanks for the explanation, I am currently doing a research based on yours, however I cannot achieve your result, mine on Garden dataset is around Recall@10: 0.1852 and Mrr: 0.0755. Could you kindly share how your hyper parameters settings?

**Comment by yzhang1918** [comment]
This is the table of hyper-parameters in a previous version. We did not include it in the CIKM'21 version due to the page limit. Though we have tweaked the model and parameters a little, I think most parameters are still within the following search spaces.

```
\begin{table*}[!b]
\caption{Hyper-parameters in \ours}
\label{tab:params}
\centering
\begin{tabular}{l|l}
\toprule
Hyper-parameter & Value \\
\midrule
Dimension $d$ & 128 \\
Number of negative samples & 16 \\
Order of Neumann series $R_{\text{inv}}$ & 10 \\
Order of Chebyshev polynomial $R_{\text{exp}}$ & 3 \\
Activation function $\sigma$ & ReLU \\
Chunk length of truncated BPTT $tbptt\_len$ & 20 \\
Frequency of meta-learning $ml\_steps$ & 5 \\
Optimizer & Adam \\
Meta-learning step size $\epsilon$ & from 1 to 0.001 (linear decay)\\
Number of inner gradient descent $inner\_steps$ & Searched from \{1, 2, 3, 4, 5 \}\\
Learning rate & Searched from \{1e-2, 2e-2, 5e-2, 1e-3, 2e-3, 5e-3, 1e-4\} \\
Weight decay & Searched from \{0, 1e-3, 1e-4, 1e-5\}\\
Coefficient of jump loss $\beta$ & Searched from \{0, 1e-2, 1e-3, 1e-4, 1e-5\}\\
\bottomrule
\end{tabular}
\end{table*}
```

**Comment by DiMarzioBian** [comment]
Thank you for the reply.

**Comment by yzhang1918** [comment]
I found a discrepancy in the evaluation part between my local and GitHub copies. I just have uploaded the new code. It should work as expected now:

```
python run_cope.py --dataset garden --cuda 0 --weight_decay 0.01
```

Sorry for the inconvenience.
